src/app/components/search/search.component.spec.ts

```diff
@@ -0,0 +1,53 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { SearchComponent } from 'src/app/components/search/search.component';
+import { ReactiveFormsModule } from '@angular/forms';
+import { ComponentsModule } from 'src/app/components/components.module';
+import { RouterTestingModule } from '@angular/router/testing';
+import { searchServiceSpy, resetSpies } from 'src/app/helpers/spies';
+import { of } from 'rxjs';
+import { NgxSpinnerModule } from 'ngx-spinner';
+
+describe('SearchComponent', () => {
+  let component: SearchComponent;
+  let fixture: ComponentFixture<SearchComponent>;
+  const values = {
+    data: {
+      data: {
+        title: 'title', description: 'description', street: 'street', state: 'state',
+        company: 'company', city: 'city', type: 'type', purchase_plan: 'F',
+        garages: 1, bathrooms: 1, bedrooms: 1, lot_size: 100, price_min: 1000, price_max: 2000,
+    }
+    }
+  };
+
+  beforeAll(() => resetSpies([searchServiceSpy]));
+  afterEach(() => resetSpies([searchServiceSpy]));
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [],
+      imports: [
+        ReactiveFormsModule,
+        RouterTestingModule,
+        ComponentsModule,
+        NgxSpinnerModule,
+      ],
+    })
+      .compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(SearchComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should use search function', () => {
+    searchServiceSpy.searchProperties.and.returnValue(of(values.data.data));
+    expect(component.search({ value: values.data.data }));
```

Comment:
Hey @JamesMudidi, this seems to always be true, regardless, I believe what we would want to do is to spy of the method and then assert if it was called, and better yet go ahead and assert what it was called with.